### PR TITLE
Show nested package references

### LINKED ISSUE #235: Ability to drill down into project references (view references of references)
Under a project > references > packages, I can see project references. I would like to be able to drill down into those to view nested dependencies, as I can in Visual Studio. Please consider implementing. Thanks.

### PR DESCRIPTION
**Pull request by badsyntax**
Fixes #235

We still list the top-level packages based on the package definitions within the .csproj file, but nested packages are discovered using the [`nuget-deps-tree`](https://www.npmjs.com/package/nuget-deps-tree) package. That package parses the `project.assets.json` file. After doing some research, this seems like the easiest approach to provide a nested package tree for nuget packages. The `project.assets.json` file should exist for dotnet core projects. 

This should be non-breaking, in that if the `nuget-deps-tree` package has any problems discovering nuget packages, then no nested package deps are displayed (but the top-level packages are still displayed).

https://user-images.githubusercontent.com/102141/208253522-8dc70204-328f-4afc-9799-28ad80c5838d.mov

### CODE REVIEW

File: src/core/Projects/MsBuildProject.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { Project as NugetProject } from 'nuget-deps-tree/dist/src/Structure/OutputStructure';
```
Comment (badsyntax) [inline src/core/Projects/MsBuildProject.ts:1]: I've created an issue to get these types exported: https://github.com/jfrog/nuget-deps-tree/issues/24

File: src/tree/items/ProjectReferencedPackageTreeItem.ts [hunk 2]
```diff
@@ -1,14 +1,41 @@
...
         this.description = pkgRef.version;
         this.allowIconTheme = false;
         this.addContextValueSuffix();
     }
 
+    protected async createChildren(childContext: TreeItemContext): Promise<TreeItem[]> {
+        let result: TreeItem[] = [];
+        const newContext = this.context.copy(this.context.project, this);
+        if (newContext.parent) {
+            newContext.parent.id = newContext.parent.id + this.pkgRef.name;
```
Comment (badsyntax) [inline src/tree/items/ProjectReferencedPackageTreeItem.ts:24]: i think this is wrong as i'm mutating `parent`. would appreciate some advice here (basically we need to support nested packages with the same package name)

**Comment by giddingsra0** [comment]
@badsyntax This is great and would be awesome to see for project references as well. Is that something easily done that could be added to this PR?

**Review by fernandoescolar — APPROVED** [review]
This PR is a great improvement but I want to use vscode library dependencies instead of external ones like `fs`. So I am going to modify the `nuget-deps-tree` dependency and write my own code to read the `project.assets.json` file.

File: src/core/Projects/MsBuildProject.ts (hunk 1 above)
Comment (fernandoescolar) [reply src/core/Projects/MsBuildProject.ts:1]: I am going to modify the `nuget-deps-tree` dependency and write my own code to read the `project.assets.json` file. So this is not a problem :)

File: src/tree/items/ProjectReferencedPackageTreeItem.ts (hunk 2 above)
Comment (fernandoescolar) [reply src/tree/items/ProjectReferencedPackageTreeItem.ts:24]: with this `id` creation code, it should not be necessary:
```ts
id = this.parent.id + '-' + this.label + '[' + this.contextValue + ']';
```

**Pull request merged**